adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/execution/InMemoryExecutionParameterRepository.java

```diff
@@ -0,0 +1,52 @@
+package org.mobilitydata.gtfsvalidator.execution;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.mobilitydata.gtfsvalidator.domain.entity.ExecutionParameter;
+import org.mobilitydata.gtfsvalidator.parser.ApacheExecutionParameterParser;
+import org.mobilitydata.gtfsvalidator.parser.JsonExecutionParameterParser;
+import org.mobilitydata.gtfsvalidator.usecase.port.ExecutionParameterRepository;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class InMemoryExecutionParameterRepository implements ExecutionParameterRepository {
```

Comment:
The expected behavior/usage of the `ExecutionParameterRepository` interface and `InMemoryExecutionParameterRepository` implementation aren't very clear to me.

For example, for a basic repo I'd expect [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) operations that all operate on the same dataset.

I would expect that I could create/initialize the repo either empty or with data via the constructor (an array of string arguments), and then add more via `addExecutionParameter()`, and query the repo when executing code to determine what I should do based on the provided parameter and value.  It should be flexible to be initialize from either file or command-line arguments.

But from what I understand, it looks like you have to initialize with a string of arguments from the command-line, which are stored internally. That, via the parser, is then added back to the repo as parsed `ExecutionParameter` to the internal `Map` representation - with the end result being a duplication of the data in both the string array as well as the map. And, then you could add more parameters via `addExecutionParameter()` and the two copies of the data would be out of sync.

It seems like `arguments` String array is just a temporary representation of the data that should be discarded, not stored in the repo.

Reply:
Absolutely! I missed that while writing the code, thanks!